src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp

```diff
@@ -223,6 +229,31 @@ namespace AppInstaller::Manifest
                 {
                     resultErrors.emplace_back(ManifestError::FieldNotSupported, "ProductId");
                 }
+
+                // Ensure that each URL has a one to one mapping with a Sha256 and
+                // warn if a Sha256 has a one to many mapping with a URL
+                if (!installer.Url.empty() && !installer.Sha256.empty())
+                {
+                    std::string checksum = Utility::SHA256::ConvertToString(installer.Sha256);
+                    std::string url = installer.Url;
+
+                    auto [urlIterator, urlInserted] = urlToChecksum.try_emplace(url, checksum);
+                    auto [checksumIterator, checksumInserted] = checksumToUrl.try_emplace(checksum, url);
+
+                    if (!urlInserted && urlIterator->second != checksum)
+                    {
+                        // If the URL was not inserted, and the value in the map does not match the current Sha256, then
+                        // a single URL corresponds to multiple SHA256 and an error should be thrown
+                        resultErrors.emplace_back(ManifestError::InconsistentInstallerHash);

```

Comment:
```suggestion
                        resultErrors.emplace_back(ManifestError::InconsistentInstallerHash, "InstallerUrl", url);
```
I see that this may have caused a build error due to ambiguity.  I would just static cast `url` to `string_view` if that is the case.

Reply:
I'll take a look. It seemed to build just fine on my end, with `ManifestInstaller::string_t` being implicitly cast to `std::string`. I can try a static cast if necessary, though